validator/validate.py

```diff
@@ -9,29 +9,29 @@
 
 
 def validate_geoparquet(input_file):
-    with open(os.path.join(sys.path[0], "schema.json"), "r") as f:
-        schema = json.loads(f.read())
-        metadata = json.loads(pq.read_schema(input_file).metadata[b"geo"])
-        errors = Draft7Validator(schema).iter_errors(metadata)
-
-        valid = True
-        print("Validating file...")
-
-        for error in errors:
-            valid = False
-            print(f"- [ERROR] {error.json_path}: {error.message}")
-            if "description" in error.schema:
-                print(f"          INFO: {error.schema['description']}")
-
-        # Extra errors
-        if (metadata["primary_column"] not in metadata["columns"]):
-            valid = False
-            print("- [ERROR] $.primary_column: must be in $.columns")
-
-        if valid:
-            print("This is a valid GeoParquet file.")
-        else:
-            print("This is an invalid GeoParquet file.")
+    schema_file = open(os.path.join(sys.path[0], "schema.json"), "r")
+    schema = json.load(schema_file)
```

Comment:
I didn't mean to take away the context manager 😅 . It's always good practice to use a context manager when opening files, but since you only use the open file to load the json data, then you can "close the context" (i.e. dedent) after reading the data. 

I.e.:
```py
from pathlib import Path

here = Path(sys.path[0])
schema_path = here / "schema.json"
with open(schema_path) as f:
    schema = json.load(f)

errors = Draft7Validator(schema).iter_errors(metadata)
...
```

Reply:
OK, thanks for the clarification :+1: